docs/src/rules/preserve-caught-error.md

```diff
@@ -0,0 +1,62 @@
+---
+title: preserve-caught-error
+rule_type: suggestion
+further_reading:
+- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause
+- https://nodejs.org/api/errors.html#errorcause
+- https://github.com/tc39/proposal-error-cause/blob/main/README.md
+- https://dev.to/amnish04/never-lose-valuable-error-context-in-javascript-3aco
+---
+
+JavaScript developers often re-throw errors in `catch` blocks to add context but forget to preserve the original error, resulting in lost debugging information.
+
+Using the `cause` option when throwing new errors helps retain the **original error** and maintain complete **error chains**, which improves debuggability and traceability.
+
+## Rule Details
+
+This rule enforces the use of the [`cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) property when throwing a new error inside a `catch` block.
+
+Checks for all built-in [error types](https://github.com/microsoft/TypeScript/blob/main/src/lib/es2022.error.d.ts) that support passing a `cause`.
```

Comment:
We typically avoid linking to other resources right in the middle of our docs. They should go in "further reading".

```suggestion
This rule enforces the use of the `cause` property in a built-error error constructor when throwing a new error inside a `catch` block.
```